### PR TITLE
bower2nix 2.1.0 -> 3.0.1

### PR DESCRIPTION
**Pull request by rvl**
This updates `bower2nix` and `fetch-bower` to fix some bugs ([release notes](https://github.com/rvl/bower2nix/releases/tag/v3.0.1)).

There is a new `buildBowerComponents` function which wraps `fetchBower` and makes it easier to use in a project.

I have also written some documentation.

cc @shlevy @rbvermaa @svanderburg 
###### Things done:
- [X] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [X] Linux
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip --against master"`.
  :question: Have run this, with a `Result in /run/user/1000/nox-review-5mnpj70p`. What is the expected output of `nox-review`?
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by shlevy** [comment]
Why is `bower2nix` not just a normal node package?

**Comment by shlevy** [comment]
Also, why are its inputs pegged by version instead of by spec respecting semver?

**Comment by rvl** [comment]
Hi @shlevy

1. `bower2nix` is not just a normal node package so that I can use `wrapProgram` to put `git` into the `PATH`. Actually the call to build with `tsc` isn't necessary so I can remove that.
2. Pegging by version seems to be what the other packages generated by `npm2nix` have, and it was easier to type.

Cheers

**Comment by shlevy** [comment]
Hmm, so I'd personally prefer to have bower2nix as a normal node package (so its dependencies are automatically managed etc.) and then have an override in `node-packages.nix` to add a `postInstall` or similar, but it's not really a huge deal and I don't maintain the node stuff anymore. @offlinehacker , thoughts?

**Comment by rvl** [comment]
Thanks @shlevy I didn't notice that it was possible to override attributes of node packages. I have updated the patch so that bower2nix is a normal node package again.

As far as I can tell, there is no way to regenerate `node-packages-generated.nix`. So the changes to this file were "semi-automated".

**Comment by shlevy** [comment]
It should be possible to update node-packages-generated with `npm2nix`, but unfortunately it's broken right now due to https://github.com/NixOS/npm2nix/issues/35.

File: doc/languages-frameworks/index.xml [hunk 1]
```diff
@@ -24,6 +24,24 @@ such as Perl or Haskell.  These are described in this chapter.</para>
...
+</section>
+
+
+<section><title>TeX / LaTeX</title>
+
+<para>* Special support for building TeX documents</para>
+
+</section>
+-->
+>>>>>>> 05113a5... nixpkgs manual: Add documentation for bower2nix
```
Comment (cstrahan) [inline doc/languages-frameworks/index.xml]: @rvl @shlevy Did this get merged in? This looks like a hunk from a bad merge.

    >>>>>>> 05113a5... nixpkgs manual: Add documentation for bower2nix

**Comment by rvl** [comment]
Ah yes I mucked up the 2nd rebase. Are you able to fix it on master?

**Pull request merged**